[processing] Fix 'equals' predicate in SpatialJoin and SpatialJoinSummary

agiudiceandrea opened the pull request:
## Description
This fixes two typos in SpatialJoin.py "Join attributes by location" and SpatialJoinSummary.py "Join attributes by location (summary)" algorithms that prevent the 'equals' predicate to work

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

agiudiceandrea committed:
Fix 'equals' predicate in SpatialJoin

This fixes a typo in SpatialJoin.py "Join attributes by location" algorithm that prevents the 'equals' predicate to work

agiudiceandrea committed:
Fix 'equals' predicate in SpatialJoinSummary

This fixes a typo in SpatialJoinSummary.py "Join attributes by location (summary)" algorithm that prevents the 'equals' predicate to work

luipir commented:
as asked by @nyalldawson, can you add a processng test for the algorithm? if already available ad one using the isEquals parameter.

agiudiceandrea committed:
Add test for Join by location (equals)

Adds a test for "Join attributes by location" processing algorithm 'equals' predicate

luipir commented:
great tnx
